src/test/java/utils/GeneralConfigModule.java

```diff
@@ -116,11 +110,11 @@ public CustomDefinition provideCustomSchemaDefinition(ResolvedType javaType, Sch
 			if (!javaType.isInstanceOf(ObjectId.class)) {
 				return null;
 			}
-			var config = context.getGeneratorConfig();
-//			var schema = config.createObjectNode().put(context.getKeyword(SchemaKeyword.TAG_TYPE), context.getKeyword(SchemaKeyword.TAG_TYPE_STRING));
-			var schema = config.createObjectNode().put(context.getKeyword(SchemaKeyword.TAG_ANYOF),
-					config.createArrayNode()
-			);
+			var schema = context.getGeneratorConfig().createObjectNode();
+			Stream.of(String.class, ObjectId.class)
+					.map(context.getTypeContext()::resolve)
+					.map(anyOfOption -> context.createStandardDefinition(anyOfOption, this))
```

Comment:
This is where the "magic" happens: the `createStandardDefinition()` method's last parameter expects this `CustomDefinitionProvider` instance as input, in order to proceed with the "normal" schema generation without applying this logic here again.
That way, the extra `anyOf` wrapper can be added without having to manually create the whole subschema.